instant-xml/src/lib.rs

```diff
@@ -181,8 +200,163 @@ pub struct FieldContext<'xml> {
     pub attribute: Option<FieldAttribute<'xml>>,
 }
 
+pub enum EntityType {
+    Element,
+    Attribute,
+}
+
+pub enum XMLTagName<'xml> {
+    FieldName,
+    Custom(&'xml str),
+}
+
 pub trait FromXml<'xml>: Sized {
-    fn from_xml(input: &str) -> Result<Self, Error>;
+    const TAG_NAME: XMLTagName<'xml>;
+
+    fn from_xml(_input: &str) -> Result<Self> {
+        unimplemented!();
```

Comment:
I think this should have a default implementation so that none of the implementers have to implement this?

Reply:
done